PipelinesTasks/PRMetrics/buildAndReleaseTask/Updaters/CodeMetrics.ts

```diff
@@ -0,0 +1,429 @@
+
+        return this.TestFactor > 0.0;
+    }
+
+    public HasSufficientTestCode(): boolean {
+        this.taskLibWrapper.debug('* CodeMetrics.HasSufficientTestCode()');
+
+        return this.SufficientTestCode;
+    }
+
+    private NormalizeParameters(
+        baseSize: string,
+        growthRate: string,
+        testFactor: string,
+        fileMatchingPatterns: string,
+        codeFileExtensions: string
+    ): void {
+        this.taskLibWrapper.debug('* CodeMetrics.NormalizeParameters()');
+
+        let integerOutput = 0;
+        integerOutput = parseInt(baseSize);
+        if (
+            isNullOrWhitespace(baseSize) ||
+            !integerOutput ||
+            integerOutput < 0
+        ) {
+            this.processWrapper.log(
+                "Write-Information -MessageData 'Adjusting base size parameter to 250.' -InformationAction 'Continue'"
+            );
+            this.BaseSize = 250;
+        } else {
+            this.BaseSize = integerOutput;
+        }
+
+        let doubleOutput = 0.0;
+        doubleOutput = parseFloat(growthRate);
+        if (
+            isNullOrWhitespace(growthRate) ||
+            !doubleOutput ||
+            doubleOutput < 1.0
+        ) {
+            this.processWrapper.log(
+                "Write-Information -MessageData 'Adjusting growth rate parameter to 2.0.' -InformationAction 'Continue'"
+            );
+            this.GrowthRate = 2.0;
+        } else {
+            this.GrowthRate = doubleOutput;
+        }
+
+        doubleOutput = parseFloat(testFactor);
+        if (
+            isNullOrWhitespace(testFactor) ||
+            !doubleOutput ||
+            doubleOutput < 0.0
+        ) {
+            this.processWrapper.log(
+                "Write-Information -MessageData 'Adjusting test factor parameter to 1.5.' -InformationAction 'Continue'"
+            );
+
+            this.TestFactor = 1.5;
+        } else {
+            this.TestFactor = doubleOutput;
+        }
+
+        if (isNullOrWhitespace(fileMatchingPatterns)) {
+            this.processWrapper.log(
+                "Write-Information -MessageData 'Adjusting file matching patterns to **/*.' -InformationAction 'Continue'"
+            );
+
+            this.FileMatchingPatterns.push('**/*');
+        } else {
+            this.FileMatchingPatterns = fileMatchingPatterns.split('\n');
+        }
+
+        this.NormalizeCodeFileExtensionsParameter(codeFileExtensions);
+    }
+
+    private NormalizeCodeFileExtensionsParameter(
+        codeFileExtensions: string
+    ): void {
+        this.taskLibWrapper.debug(
+            '* CodeMetrics.NormalizeCodeFileExtensionsParameter()'
+        );
+
+        if (isNullOrWhitespace(codeFileExtensions)) {
+            this.processWrapper.log(
+                "Write-Information -MessageData 'Adjusting code file extensions parameter to default values.' -InformationAction 'Continue''"
+            );
+
+            this.CodeFileExtensions = [
```

Comment:
Can you add .js and .ts to this list? We had a request to get these added so it would be great to have them now